### PR TITLE
Cancel outstading requests when the application is rejected

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

After accepting an offer a candidate can still end up in an unsuccessful end state. Then we will cancel the references.

For now, this PR adds only for the rejected state.

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[Cancel outstanding references if the candidate ends up in an unsuccessful state](https://trello.com/c/S2abbrCV/554-cancel-outstanding-references-if-the-candidate-ends-up-in-an-unsuccessful-state)

File: app/services/cancel_outstanding_references.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class CancelOutstandingReferences
+  attr_reader :application_form
+
+  delegate :application_references, :show_new_reference_flow?, to: :application_form
+
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call!
```
Comment (carlosmartinez) [inline app/services/cancel_outstanding_references.rb:10]: Do we have a convention where `!` indicates something's getting deleted?

File: spec/services/reject_application_spec.rb [hunk 2]
```diff
@@ -93,5 +93,15 @@
       service.save
       expect(cancel_upcoming_interviews).to have_received(:call!)
     end
+
+    it 'calls the CancelOutstandingReferences service' do
+      cancel_outstanding_references = instance_double(CancelOutstandingReferences, call!: true)
+
+      allow(CancelOutstandingReferences)
+        .to receive(:new).with(application_form: application_choice.application_form)
+                         .and_return(cancel_outstanding_references)
```
Comment (carlosmartinez) [inline spec/services/reject_application_spec.rb:102]:
```
      allow(CancelOutstandingReferences)
        .to receive(:new)
        .with(application_form: application_choice.application_form)
        .and_return(cancel_outstanding_references)
```
por favor

**Review by carlosmartinez — APPROVED** [review]

File: app/services/cancel_outstanding_references.rb (hunk 1 above)
Comment (tomas-stefano) [reply app/services/cancel_outstanding_references.rb:10]: We do have some services like cancel interviews that uses

File: spec/services/reject_application_spec.rb (hunk 2 above)
Comment (tomas-stefano) [reply spec/services/reject_application_spec.rb]: rubocop did this identation 🤷 but I am happy to change!

Comment (tomas-stefano) [reply spec/services/reject_application_spec.rb]: Done!

Comment (tomas-stefano) [reply spec/services/reject_application_spec.rb]: Obrigado!

**Review by JR-G — APPROVED** [review]
🚀

**Pull request merged**